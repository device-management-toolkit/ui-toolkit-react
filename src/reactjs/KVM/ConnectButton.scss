/*********************************************************************
 * Copyright (c) Intel Corporation 2019
 * SPDX-License-Identifier: Apache-2.0
 **********************************************************************/

.button{
    margin-right: 3px;
    padding: 2px;
    min-width: 90px
}