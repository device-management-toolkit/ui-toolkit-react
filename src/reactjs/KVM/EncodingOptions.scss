/*********************************************************************
 * Copyright (c) Intel Corporation 2019
 * SPDX-License-Identifier: Apache-2.0
 **********************************************************************/
 
.encoding {
    margin-left: 10px;
  }
  .encoding label {      
    padding: 2px;
    font-size: 15px;
  }
  .encoding select {
    padding: 2px 0;
  }
  
  .reldisabled{
    opacity: 0.5;
  }