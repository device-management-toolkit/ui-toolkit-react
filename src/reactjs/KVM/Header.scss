/*********************************************************************
 * Copyright (c) Intel Corporation 2019
 * SPDX-License-Identifier: Apache-2.0
 **********************************************************************/
 
.header {
  background-color: darkgray;
  padding: 5px;
  font-size: 13px;
}
