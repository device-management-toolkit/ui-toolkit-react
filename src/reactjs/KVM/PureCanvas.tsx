/*********************************************************************
 * Copyright (c) Intel Corporation 2019
 * SPDX-License-Identifier: Apache-2.0
 * Author : Ramu Bachala
 **********************************************************************/

import React from 'react'
import { isFalsy } from '../shared/Utilities'
import './PureCanvas.scss'

export interface PureCanvasProps {
  contextRef: (ctx: CanvasRenderingContext2D) => void
  mouseDown: (event: React.MouseEvent) => void
  mouseUp: (event: React.MouseEvent) => void
  mouseMove: (event: React.MouseEvent) => void
  canvasHeight: string
  canvasWidth: string
}

export class PureCanvas extends React.Component<PureCanvasProps, {}> {
  shouldComponentUpdate (): boolean {
    return false
  }

  render (): React.ReactNode {
    const canvasAttributes: React.CanvasHTMLAttributes<HTMLCanvasElement> = {
      width: '1366',
      height: '768',
      onContextMenu: (e) => { e.preventDefault(); return false },
      onMouseDown: this.props.mouseDown,
      onMouseUp: this.props.mouseUp,
      onMouseMove: this.props.mouseMove
    }
    return (
      <canvas {...canvasAttributes} className="canvas" ref={(c: any) => isFalsy(c) ? this.props.contextRef(c.getContext('2d')) : null}/>
    )
  }
}
