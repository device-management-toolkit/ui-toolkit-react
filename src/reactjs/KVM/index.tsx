/*********************************************************************
* Copyright (c) Intel Corporation 2019
* SPDX-License-Identifier: Apache-2.0
**********************************************************************/
import React from 'react'
import { render } from 'react-dom'
import { KVM } from './UI'
import i18n from '../../i18n'
// Get browser language
i18n.changeLanguage(navigator.language).catch(() => console.info('error occured'))

const url = new URL(window.location.href)
const params = new URLSearchParams(url.search)

render(<KVM autoConnect={false} deviceId={params.get('deviceId')} mpsServer={params.get('mpsServer') + '/relay'} authToken="authorization" mouseDebounceTime={200} canvasHeight={'100%'} canvasWidth={'100%'} />, document.querySelector('#kvm'))
