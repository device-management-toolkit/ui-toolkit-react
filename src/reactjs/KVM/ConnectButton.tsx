/*********************************************************************
 * Copyright (c) Intel Corporation 2019
 * SPDX-License-Identifier: Apache-2.0
 **********************************************************************/

import React from 'react'
import './ConnectButton.scss'

export interface ConnectProps {
  kvmstate: number
  handleConnectClick: (e: any) => void
}

export class ConnectButton extends React.Component<ConnectProps, {}> {
  render (): React.ReactNode {
    return (
      <button className="button" onClick={this.props.handleConnectClick}>
        {this.props.kvmstate === 1 ? 'Connecting KVM' : (this.props.kvmstate === 2 ? 'Disconnect KVM' : 'Connect KVM')}
      </button>
    )
  }
}
