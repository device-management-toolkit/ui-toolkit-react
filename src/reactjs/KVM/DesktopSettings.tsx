/*********************************************************************
 * Copyright (c) Intel Corporation 2019
 * SPDX-License-Identifier: Apache-2.0
 **********************************************************************/

import React from 'react'
import { EncodingOptions } from './EncodingOptions'

export interface IDesktopSettings {
  changeDesktopSettings: (settings: any) => void
  getConnectState: () => number
}

export class DesktopSettings extends React.Component<IDesktopSettings> {
  desktopsettings = {
    encoding: 1
  }

  constructor (props: IDesktopSettings) {
    super(props)
    this.changeEncoding = this.changeEncoding.bind(this)
  }

  changeEncoding (encoding: number): void {
    this.desktopsettings.encoding = encoding
    this.props.changeDesktopSettings(this.desktopsettings)
  }

  render (): React.ReactNode {
    return (
      <EncodingOptions changeEncoding={this.changeEncoding} getConnectState={this.props.getConnectState}/>
    )
  }
}
