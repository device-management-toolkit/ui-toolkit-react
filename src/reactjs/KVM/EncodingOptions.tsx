/*********************************************************************
 * Copyright (c) Intel Corporation 2019
 * SPDX-License-Identifier: Apache-2.0
 **********************************************************************/

import React from 'react'
import './EncodingOptions.scss'

export interface IEncodingOptions {
  changeEncoding: (encoding: number) => void
  getConnectState: () => number
}

export class EncodingOptions extends React.Component<IEncodingOptions, { value: number }> {
  constructor (props: IEncodingOptions) {
    super(props)
    this.state = { value: 1 }
    this.onEncodingChange = this.onEncodingChange.bind(this)
  }

  onEncodingChange (e): void {
    // e.persist();
    this.setState({ value: e.target.value })
    this.props.changeEncoding(e.target.value)
  }

  render (): React.ReactNode {
    return (
      <span className="encoding">
        <label >Encoding:</label>
        <select value={this.state.value} className= {this.props.getConnectState() === 2 ? 'reldisabled' : ''} onChange={this.onEncodingChange} disabled={this.props.getConnectState() === 2}>
          <option value="1">RLE 8</option>
          <option value="2">RLE 16</option>
        </select>
      </span>
    )
  }
}
