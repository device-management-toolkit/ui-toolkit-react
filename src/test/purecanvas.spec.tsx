/*********************************************************************
* Copyright (c) Intel Corporation 2019
* SPDX-License-Identifier: Apache-2.0
**********************************************************************/

import React from 'react'
import { PureCanvas, PureCanvasProps } from '../reactjs/KVM/PureCanvas'
import { shallow } from 'enzyme'
import { any } from 'prop-types'

const testprops: PureCanvasProps = {
  contextRef: (ctx: CanvasRenderingContext2D) => {},
  mouseDown: (event: React.MouseEvent) => {},
  mouseUp: (event: React.MouseEvent) => {},
  mouseMove: (event: React.MouseEvent) => {},
  canvasHeight: ('768'),
  canvasWidth: ('1366')
}

describe('Testing purecanvas component rendering', () => {
  it('renders without crashing', () => {
    shallow(<PureCanvas {...testprops} />)
  })
})

describe('Testing purecanvas component update', () => {
  it('component update should return false', () => {
    const wrapper: any = shallow(<PureCanvas {...testprops} />)
    const result = wrapper.instance()?.shouldComponentUpdate(any, any, any)
    expect(result).toBe(false)
  })
})

describe('Testing render method', () => {
  it('render() width testing', () => {
    const wrapper = shallow(<PureCanvas {...testprops} />)
    expect(wrapper.prop('width')).toBe('1366')
  })
})

describe('Testing render method', () => {
  it('render() height testing', () => {
    const wrapper = shallow(<PureCanvas {...testprops} />)
    expect(wrapper.prop('height')).toBe('768')
  })
})

//
describe('Test PureCanvas', () => {
  it('Test render() in PureCanvas', () => {
    // Initialization of PureCanvasProps
    const purecanvasprops: PureCanvasProps = {
      contextRef: (ctx: CanvasRenderingContext2D) => {},
      mouseDown: (event: React.MouseEvent) => {},
      mouseUp: (event: React.MouseEvent) => {},
      mouseMove: (event: React.MouseEvent) => {},
      canvasHeight: ('700'),
      canvasWidth: ('300')
    }
    let myValue = 0
    const myEvent = {
      preventDefault () { myValue = 1 }
    }

    // test render and simulate event
    const pc = shallow(<PureCanvas {...purecanvasprops} />)
    pc.simulate('contextmenu', myEvent)

    // Output
    expect(pc.prop('onMouseDown')).toBe(purecanvasprops.mouseDown)
    expect(pc.prop('onMouseUp')).toBe(purecanvasprops.mouseUp)
    expect(pc.prop('onMouseMove')).toBe(purecanvasprops.mouseMove)
    expect(pc.prop('width')).toBe('1366')
    expect(pc.prop('height')).toBe('768')
    expect(myValue).toBe(1)
    expect(pc).toMatchSnapshot()
    console.log(pc.debug())
    console.log(pc.props())
  })
})
