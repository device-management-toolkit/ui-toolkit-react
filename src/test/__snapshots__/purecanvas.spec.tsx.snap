// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PureCanvas Test render() in PureCanvas 1`] = `ShallowWrapper {}`;
